bin/tictactoe

```diff
@@ -1,3 +1,34 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+puts "Welcome to Tic Tac Toe!"
+puts "Enter 1 for single player, 2 for head to head, or 0 for AI mode"
+
+num = gets.chomp
+
+case num
+
+  when "1"
+    puts "enter 1 to go first, or 2 to go second"
+    input = gets.chomp
+    if input == "1"
+      player_1 = Player::Human.new("X")
+      player_2 = Player::Computer.new("O")
+    else
+      player_1 = Player::Computer.new("X")
+      player_2 = Player::Human.new("O")
+    end
+  when "2"
+    player_1 = Player::Human.new("X")
+    player_2 = Player::Human.new("O")
+  when "0"
+    player_1 = Player::Computer.new("X")
+    player_2 = Player::Computer.new("O")
+  else
+    puts "Sorry, try again."
+end
+
+Game.new(player_1, player_2).play
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.